src/sentry/static/sentry/app/stores/configStore.tsx

```diff
@@ -28,7 +29,10 @@ const configStoreConfig: Reflux.StoreDefinition & ConfigStoreInterface = {
   },
 
   set(key, value) {
-    this.config[key] = value;
+    this.config = {
+      ...this.config,
+      [key]: value,
+    };
```

Comment:
Previously, we were mutating the config object which is not ideal as it can cause issues with react lifecycle methods (eg shallow comparing props). The `withConfig` test illustrates this as the object is mutated and the value is changed without an explicit component update.